pkg/context/txnclient/client.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package txnclient
+
+import (
+	"sync"
+
+	"github.com/trustbloc/sidetree-core-go/pkg/api/operation"
+	txnapi "github.com/trustbloc/sidetree-core-go/pkg/api/txn"
+
+	"github.com/trustbloc/orb/pkg/didtxnref"
+	"github.com/trustbloc/orb/pkg/txngraph"
+)
+
+// Client implements writing orb transactions.
+type Client struct {
+	namespace string
+	txnGraph  txnGraph
+	didTxns   didTxns
+	txnCh     chan []string
+	sync.RWMutex
+}
+
+type txnGraph interface {
+	Add(info *txngraph.Node) (string, error)
+}
+
+type didTxns interface {
+	Add(did, cid string) error
+	Get(did string) ([]string, error)
+}
+
+// New returns a new orb transaction client.
+func New(namespace string, graph txnGraph, txns didTxns, txnCh chan []string) *Client {
+	return &Client{
+		namespace: namespace,
+		txnGraph:  graph,
+		didTxns:   txns,
+		txnCh:     txnCh,
+	}
+}
+
+// WriteAnchor writes anchor string to orb transaction.
+func (c *Client) WriteAnchor(anchor string, refs []*operation.Reference, version uint64) error {
+	// assemble map of previous did transaction for each did that is referenced in anchor
+	previousDidTxns := make(map[string]string)
+
+	for _, ref := range refs {
+		txns, err := c.didTxns.Get(ref.UniqueSuffix)
```

Comment:
I assume this returns an ordered list of transactions?

Reply:
yes